ffc-pay-event/save-projection.js

```diff
@@ -0,0 +1,13 @@
+const saveProjection = (context, event) => {
+  const frn = event?.action?.data?.paymentRequest?.frn ?? event?.action?.data?.frn
+  const paymentRequestNumber = event?.action?.data?.paymentRequest?.paymentRequestNumber ?? event?.action?.data?.paymentRequestNumber
+  const agreementNumber = event?.action?.data?.paymentRequest?.agreementNumber ?? event?.action?.data?.agreementNumber
+  if (frn && paymentRequestNumber && agreementNumber) {
+    context.log.info(`Creating projection for ${event.id} with frn ${frn}, paymentRequestNumber ${paymentRequestNumber}, agreementNumber ${agreementNumber}`)
+    context.bindings.outputSbTopic = { id: event.id, frn, paymentRequestNumber, agreementNumber }
+  } else {
+    context.log.info(`Skipping projection for ${event.id} as mandatory projection data not present`)
+  }
+}
+
+module.exports = saveProjection
```

Comment:
Just moving as is.  I think there's definite log term scope for tidying this up though when we rebuild in k8s.

Reply:
i?.agree?.on?.that?.point?.or?.,?.do?.i?